code-challenges.js

```diff
@@ -7,60 +7,120 @@
 // Use the test variable provided below. Expected outcome: [40, -35, 0, 30, 10]
 
-var myNumbers2 = [8, -7, 0, 6, 2]
-
-
+const map5 = (array) => {
+  return array.map(value => value * 5);
+}
 
+var myNumbers2 = [8, -7, 0, 6, 2]
 
+// Test
+console.log(map5(myNumbers2), "\n");
 
 // --------------------4) Create a function that removes all the vowels from a string.
 // Use the test variables provided below. Expected output: "HyThrLrnStdnt" "LvJvScrpt"
 
+const byeVowels = (string) => {
+  let newArr = string.split("")
+  let filteredArr = newArr.filter(value => {
+    return value.toLowerCase() !== "a" && value.toLowerCase() !== "e" && value.toLowerCase() !== "i" && value.toLowerCase() !== "o" && value.toLowerCase() !== "u"
+  })
+  return filteredArr.join("")
+}
+
 var stringWithVowels1 = "HeyThereLearnStudent"
 var stringWithVowels2 = "ILoveJavaScript"
 
-
-
-
+// Tests
+console.log(byeVowels(stringWithVowels1));
+console.log(byeVowels(stringWithVowels2), "\n");
 
 // --------------------5) Copy/paste your code from #4. Refactor your code to include non-string edge cases. Inform your user that the variable passed into the vowel removal function is not a string.
 // Use the test variables provided below. Expected output: "true is not a string" "42 is not a string"
 
+const betterByeVowels = (string) => {
+  if(typeof string === "string"){
+    return string.replace(/[aeiou]/ig, "")
+  }
+  else {
+    return `${ string } is not a string`
+  }
+}
+
 var notAString1 = true
 var notAString2 = 42
 
-
-
-
+//Tests
+console.log(betterByeVowels(stringWithVowels1));
+console.log(betterByeVowels(stringWithVowels2));
+console.log(betterByeVowels(notAString1));
+console.log(betterByeVowels(notAString2), "\n");
 
 // --------------------6) Create a function that takes the toonimals array and returns an array with only the toon objects that are cats.
 // Expected output: [ { name: "Stimpy", animal: "cat" }, { name: "Scratchy", animal: "cat" }, { name: "Felix", animal: "cat" } ]
 
-var toonimals = [ { name: "Itchy", animal: "mouse" }, { name: "Stimpy", animal: "cat" }, { name: "Daffy", animal: "duck" }, { name: "Scratchy", animal: "cat" }, { name: "Ren", animal: "dog" }, { name: "Felix", animal: "cat" }]
-
-
+const onlyCats = (array) => {
+  return array.filter(value => value.animal === "cat")
+}
 
+var toonimals = [ { name: "Itchy", animal: "mouse" }, { name: "Stimpy", animal: "cat" }, { name: "Daffy", animal: "duck" }, { name: "Scratchy", animal: "cat" }, { name: "Ren", animal: "dog" }, { name: "Felix", animal: "cat" }]
 
+// Test
+console.log(onlyCats(toonimals), "\n");
 
 // --------------------7) Using the toonimals variable from #6, create a function that returns only the names of the non-cats.
 // Expected output: "Itchy" "Daffy" "Ren"
+
+const nonCats = (array) => {
+  let filteredArr = array.filter(value => value.animal !== "cat")
+  return filteredArr.map(value => value.name)
```

Comment:
This is totally excellent. I think there may be a way you could refactor this to use only the map with a conditional statement inside of its block of code to do the exact same thing. I'd look into this!